### PR TITLE
v3 example changes

### PR DESCRIPTION
**Pull request by FullstackGJJ**

### CODE REVIEW

File: examples/fi_example.py
Comment (saigiridhar21) [inline examples/fi_example.py:55]: Can you pass all the values as named arguments?

File: examples/pa_engine_single_unit_example.py
Comment (saigiridhar21) [inline examples/pa_engine_single_unit_example.py:68]: Can you please pass the deadline header value here?

File: examples/pa_engine_single_unit_example.py
Comment (saigiridhar21) [inline examples/pa_engine_single_unit_example.py:71]: Can you add a check on the `200` status code here?

**Review by saigiridhar21 — COMMENTED** [review]
Left some comments.

File: examples/axp_optimizer_example.py
Comment (AkankshaAcharya) [inline examples/axp_optimizer_example.py:62]: Can you use variables to store the values and pass them here similar to pa examples?

File: examples/pa_engine_single_unit_example.py
Comment (FullstackGJJ) [reply examples/pa_engine_single_unit_example.py:68]: What value would you want it to be set as in this example?

File: examples/axp_optimizer_example.py
Comment (a-ghag) [reply examples/axp_optimizer_example.py:62]: hey Akanksha, in general I always prefer adding variables but these variables are not adding much value since they are only used in one location. I was irritated myself with these variables as they were just unnecessarily adding time/complexity to build examples. I only did it for consistency. But if we all are fine we can start skipping them going forward for all examples.

File: examples/pa_engine_single_unit_example.py
Comment (FullstackGJJ) [reply examples/pa_engine_single_unit_example.py:68]: I'll just do 30 seconds

File: examples/pa_engine_single_unit_example.py
Comment (FullstackGJJ) [reply examples/pa_engine_single_unit_example.py:71]: I added it to the multiple unit example where 200 status code is returned for it giving an aggregate calculation to keep track of.

File: examples/pa_engine_single_unit_example.py
Comment (a-ghag) [reply examples/pa_engine_single_unit_example.py:68]: @saigiridhar21 since client usually copy paste these examples in their code should we add the headers to examples? I am afraid they will permanently add it even if they don't need it. Would be better if clients use the default value instead of they hard coding it here?

File: examples/pa_engine_single_unit_example.py
Comment (AkankshaAcharya) [reply examples/pa_engine_single_unit_example.py:71]: @a-ghag we return 200 when the calculation completes with an error for a single unit

File: examples/pa_engine_multiple_unit_example.py
Comment (AkankshaAcharya) [inline examples/pa_engine_multiple_unit_example.py:66]: Can we add meta value here as they are configurable by the client?

File: examples/pa_engine_single_unit_example.py
Comment (AkankshaAcharya) [reply examples/pa_engine_single_unit_example.py:68]: @a-ghag we could just assign the default values to the header in the code and mention the same in a comment for the clients' information. Would there be an issue this way?

Comment (a-ghag) [reply examples/pa_engine_single_unit_example.py:68]: So if we assign the the same default value as in our code here and later we change the default value in our code clients would still be using the old default as they would be using the hard coded value from these examples. Is it good to assume that we did this in Java? If yes, can we remove those?

File: examples/pa_engine_single_unit_example.py
Comment (a-ghag) [reply examples/pa_engine_single_unit_example.py:71]: Oh good point. I keep forgetting about it.

File: examples/pa_engine_multiple_unit_example.py
Comment (a-ghag) [reply examples/pa_engine_multiple_unit_example.py:66]: I think it makes sense to skip setting the meta values as we would want clients to use the defaults from our code. If we have done this in Java can we remove those values?

File: examples/pa_engine_single_unit_example.py
Comment (a-ghag) [reply examples/pa_engine_single_unit_example.py:71]: done

File: examples/pa_engine_single_unit_example.py
Comment (saigiridhar21) [reply examples/pa_engine_single_unit_example.py:68]: @a-ghag I am fine removing `cache-control` from the examples. Shall we document it in PDF documentation? That way clients can refer to it there.
@AkankshaAcharya Please remove the cache-control option when you work on Java SDK next time.

File: examples/pa_engine_multiple_unit_example.py
Comment (saigiridhar21) [reply examples/pa_engine_multiple_unit_example.py:66]: Agrees. Let's remove it.

File: examples/axp_optimizer_example.py
Comment (saigiridhar21) [inline examples/axp_optimizer_example.py:83]: Is `_return_http_data_only` false by default?

File: examples/fi_example.py
Comment (saigiridhar21) [inline examples/fi_example.py:88]: Shouldn't it be `run_calculation_response[0].data.calculationid`?

File: examples/fi_example.py
Comment (saigiridhar21) [inline examples/fi_example.py:85]: Shouldn't it be `run_calculation_response[0].data`?

File: examples/requirements.txt
Comment (saigiridhar21) [inline examples/requirements.txt:5]: Can you please make it `<2.0.0.` for all the 3 packages?

File: examples/requirements.txt
Comment (saigiridhar21) [inline examples/requirements.txt:2]: Can you please make it `<6.0.0`?

File: examples/axp_optimizer_example.py
Comment (FullstackGJJ) [reply examples/axp_optimizer_example.py:83]: No, it's true by default

File: examples/requirements.txt
Comment (FullstackGJJ) [reply examples/requirements.txt:5]: Yes

File: examples/requirements.txt
Comment (FullstackGJJ) [reply examples/requirements.txt:2]: Yes

File: examples/fi_example.py
Comment (FullstackGJJ) [reply examples/fi_example.py:88]: It didn't display that behavior when I ran and tested, is that a problem?

Comment (FullstackGJJ) [reply examples/fi_example.py:88]: No, because the model that gets generated `CalculationInfo` has that field being `id` https://github.com/factset/analyticsapi-engines-python-sdk/blob/140b36ca933e15a4887b37565668bf2a72d842e1/auto-generated-sdk/fds/analyticsapi/engines/model/calculation_info.py#L85

File: examples/axp_optimizer_example.py
Comment (saigiridhar21) [inline examples/axp_optimizer_example.py:127]: For optimizers, we need to run a for-loop on the tables list. Can you please refer to Java and implement the same here?

**Review by saigiridhar21 — COMMENTED** [review]
Looks good except for one comment.

File: examples/axp_optimizer_example.py
Comment (FullstackGJJ) [reply examples/axp_optimizer_example.py:127]: I was looking at the code for https://github.com/factset/analyticsapi-engines-java-sdk/blob/master/examples/src/examples/BpmInteractiveOptimizerEngineExample.java#L145. I don't know how this (https://github.com/factset/analyticsapi-engines-python-sdk/pull/34/files/d18826bfc95b9ffff31d8f26ec04b85a29bcce0b#diff-9a660acba24f88fd0f1c5cf8ed7ec9e86b79b804bc7c3a0bd715c929bedad2dbR130) differs from that.

File: examples/pa_engine_single_unit_example.py
Comment (a-ghag) [reply examples/pa_engine_single_unit_example.py:72]: Rose is working on a document which goes in detail about about cache-control directive

File: examples/axp_optimizer_example.py
Comment (a-ghag) [reply examples/axp_optimizer_example.py:88]: @saigiridhar21 is it worth make this change in open api generator?

File: examples/axp_optimizer_example.py
Comment (a-ghag) [reply examples/axp_optimizer_example.py]: done

File: examples/axp_optimizer_example.py
Comment (saigiridhar21) [reply examples/axp_optimizer_example.py:90]: yes. Considering that we return multiple status codes, it is better to make the change.

File: examples/axp_optimizer_example.py
Comment (saigiridhar21) [inline examples/axp_optimizer_example.py:23]: Can you please change the environment variable to `ANALYTICS_API_QA_USERNAME_SERIAL`?

File: examples/axp_optimizer_example.py
Comment (saigiridhar21) [inline examples/axp_optimizer_example.py:128]: If I am correct, it should be `result['optimal']`.

File: examples/quant_engine_single_unit_example.py
Comment (saigiridhar21) [inline examples/quant_engine_single_unit_example.py:90]: As this is STACH response, shouldn't it be `result_response[0].data`?

**Review by saigiridhar21 — APPROVED** [review]
Left some minors. Please address them before merging these changes.

File: examples/pa_engine_single_unit_example.py
Comment (saigiridhar21) [inline examples/pa_engine_single_unit_example.py:59]: In Java, I think that we tried covering all the possible POST body fields like for example, `holdingsMode` here. If possible, add them, or please create a JIRA card.

File: examples/axp_optimizer_example.py
Comment (FullstackGJJ) [reply examples/axp_optimizer_example.py:23]: A person reading this code has the perspective of them being a user, their concept of QA would differ from our concept of QA. I worry that breaks abstraction.

**Comment by AkankshaAcharya** [comment]
@FullstackGJJ will you be adding the AFI example in this PR? Or would you prefer me doing that in a separate PR?

File: examples/pa_engine_single_unit_example.py
Comment (FullstackGJJ) [reply examples/pa_engine_single_unit_example.py:59]: JIRA task created as follow up change.

**Comment by FullstackGJJ** [comment]
@AkankshaAcharya I will be adding right now

File: examples/afi_optimizer_example.py
Comment (a-ghag) [inline examples/afi_optimizer_example.py:106]: Can you change `result['trades']` to `result['optimal']`

File: examples/axp_optimizer_example.py
Comment (a-ghag) [inline examples/axp_optimizer_example.py:115]: I think this needs to be `status_response[0].errors`. You will have to make the change in other locations as well

File: examples/bpm_optimizer_example.py
Comment (a-ghag) [inline examples/bpm_optimizer_example.py:106]: Can you change this to optimal

File: examples/fpo_optimizer_example.py
Comment (a-ghag) [inline examples/fpo_optimizer_example.py:116]: errors

File: examples/fpo_optimizer_example.py
Comment (a-ghag) [inline examples/fpo_optimizer_example.py:129]: optimal

File: examples/afi_optimizer_example.py
Comment (a-ghag) [inline examples/afi_optimizer_example.py:93]: errors

File: examples/bpm_optimizer_example.py
Comment (a-ghag) [inline examples/bpm_optimizer_example.py:93]: errors

**Review by a-ghag — APPROVED** [review]
Minor issues but looks good to merge once those are fixed

**Pull request merged**
